### ISSUE TITLE
[Feature Request] Annotate steps with human readable info

### ISSUE THREAD

**Issue by Fishbowler** [mobile-dev-inc/maestro#1229]
I'd love to be able to put some descriptive text against my steps to make the intent of some steps clearer to the reader. 

I can do comments, but being able to do something like this is useful in certain situations where we can be can't be more contextual with testIDs and the like.

```yaml
- tapOn:
    point: "75%,78%"
    label: "Tap the native back button"
```

Not only does this help future maintenance, it could be brought into the output. A made up example below shows an improved output where items with `<---` have a label thats displayed in place of what would've been printed. 

```
 ║  > Flow                                                                      
 ║                                                                              
 ║    ✅  Launch app "com.example"                                          
 ║    ✅  Tap on "Sign in"                                                      
 ║    ✅  Tap on id: loginEmail                                                 
 ║    ✅  Input username of an Administrative user <---
 ║    ✅  Tap on id: loginPassword                                              
 ║    ✅  Input password for an Administrative user  <---             
 ║    ✅  Press Enter key                                                       
 ║    ✅  Tap on "Search content"
 ║    ✅  Input text test                                            
 ║    ✅  Tap on the first test result <---
 ║    ✅  Tap on the back button <---
```

**Comment by SuperRoach** [comment]
I'm interested in taking this one on, but I've never done kotlin before. I can see that there is a renderCommands, printLogMessages, and a renderCommandsPlainText that would need to be updated, which take state from a list.
The actual rendering of the line happens here for Ansi: 

https://github.com/mobile-dev-inc/maestro/blob/d39265d574881cc18a36ebf1b4595db35e982baf/maestro-cli/src/main/java/maestro/cli/runner/resultview/AnsiResultView.kt#L118

And I could do a ternary statement where I render a "label" if present, otherwise the usual way as a fallback.

**Comment by Fishbowler** [comment]
Another option is to change or add to the [MaestroCommand.description method](https://github.com/mobile-dev-inc/maestro/blob/main/maestro-orchestra-models/src/main/java/maestro/orchestra/MaestroCommand.kt#L145-L147), and to fetch the label if it's set, else fetch the description. Change vs add depends on where else description() is used?

Disclaimer: I've not touched Kotlin in a few years, and I've not yet touched the Maestro codebase at all, but trying to help for entirely self-serving purposes 😁

**Comment by SuperRoach** [comment]
No actually that helps, I can see that all commands have a description that is used. I could add overrides for some of the common ones as a starter. 

I've since found I can run maestro locally with ./maestro, albeit on the machine I have right now I can't run ADB via wsl2, and the web experimental view hangs even with chromedriver installed...

**Comment by SuperRoach** [comment]
As a follow up I'm not sure if "label" is the best option, but I've run with it for now. I've got a Draft PR that I'll want to be able to test locally before I run it by the team.

https://github.com/mobile-dev-inc/maestro/pull/1238

There are a fair few commands I didn't add the label too, more out of convenience for me and they seem like they wouldn't be the kind of command you wouldn't want a label on anyway.

**Comment by Fishbowler** [comment]
Wow! That's some awesome quick work @SuperRoach!

**Comment by SuperRoach** [comment]
I thought of a main use case for this as well. 

When recording a demo, or showing it to a client, it's sometimes helpful to hide an inputText step that is putting in a password. You can specify in the label  to instead say something like "Enter an ****" or "Entering password" instead of the actual text.

**Comment by Fishbowler** [comment]
Do you need any help testing? I've now got Maestro checked out and running on devices, so if you can give pointers on flows, I can happily lend a hand

**Comment by Fishbowler** [comment]
I've popped some comments on PR - I've been able to move it forward, but I haven't tackled InputText. Because it didn't previously have any parameters, it's going to require some YamlInputUnion or something to deal with "I provided some text" vs "I provided a YAML object that includes text, and maybe a label"

**Comment by Fishbowler** [comment]
Interesting question: How should the JUnit reporter behave?

**Comment by Fishbowler** [comment]
Fixed in #1292
